### PR TITLE
feat(docs): Add documentation on upgrading from Node 6

### PR DESCRIPTION
**Pull request by m-allanson**
## Description

Add docs on Gatsby's Node support policy, and how to upgrade from Node 6
<details>
<summary>View screenshot </summary>

localhost_8000_docs_upgrading-node-js (4)

</details>

## Related Issues

Related #14231

### CODE REVIEW

File: docs/docs/upgrading-node-js.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# Upgrading your Node.js version
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]:
```suggestion
---
title: Upgrading your Node.js version
---
```

File: docs/docs/upgrading-node-js.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## Introduction
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]: Probably just remove?

```suggestion
```

File: docs/docs/upgrading-node-js.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+## What version of Node do I have?
+
+Run `node -v` in a terminal to see which version of Node you have.
+
+```
+node -v
+v10.16.0
+```
+
+This example shows Node version 10.
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]: Un-opinionated on this 🤷‍♂ could be worth doing. Up to you!

```suggestion
This example shows Node version 10, specifically v10.16.0.
```

File: docs/docs/upgrading-node-js.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+## Upgrading from Node version 6
+
+Node version 6 reached _End-of-life_ status on 30th April 2019. Many of Gatsby's dependencies are updating to Node version 8 and above. Gatsby must also update in order release new features and bug fixes.
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]:
```suggestion
Node version 6 reached _End-of-life_ status on 30th April 2019. Many of Gatsby's dependencies are updating to Node version 8 and above. Gatsby must also update in order to deliver new features and bug fixes more quickly.
```

File: docs/docs/upgrading-node-js.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+Generally we'd recommend using [the Node version who's status is _Active LTS_](https://github.com/nodejs/Release#nodejs-release-working-group) (Node 10 at time of writing). However, in this document you'll learn how to update from Node 6 to Node 8 as this is likely to be the least disruptive upgrade for you.
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]:
```suggestion
Generally we'd recommend using [the Node version whose status is _Active LTS_](https://github.com/nodejs/Release#nodejs-release-working-group) (Node 10 at time of writing). However, in this document you'll learn how to update from Node 6 to Node 8 as this is likely to be the least disruptive upgrade for you.
```

File: docs/docs/upgrading-node-js.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+```
+n
+```
+
+in a terminal to see if n is installed on your system. If it's installed, you can run `n 8` to install and use Node version 8.
+
+[Check n's documentation for further instructions](https://github.com/tj/n).
+
+#### nvm
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]: Nit: but I'd maybe move nvm above n? Perhaps it's just my bubble--but it seems that nvm is much more heavily used.

File: docs/docs/upgrading-node-js.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+to install and use Node version 8.
+
+[Check nvm's documentation for further instructions](https://github.com/nvm-sh/nvm).
+
+### Installing from nodejs.org
+
+If you aren't using any of the previously listed installation methods, you can [download a Node installer directly from nodejs.org](https://nodejs.org/en/).
+
+Gatsby's recommended way to install Node is by using Homebrew. See the Homebrew section of this document for more info.
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]:
```suggestion
Gatsby's recommended way to install Node is by using Homebrew. Refer to the previous [Homebrew section of this document](#using-homebrew) for more info.
```

File: docs/docs/upgrading-node-js.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+## Conclusion
+
+Gatsby takes backwards compatibility seriously and aims to support older versions of Node for as long as possible. We understand that juggling different software versions is not a productive way to spend your day.
+
+Gatsby also relies on a huge ecosystem of JavaScript dependencies. As the ecosystem moves away from older, unsupported Node versions we have to keep pace to ensure that bugs can be fixed and new features can be released.
+
+In this document you learned how you upgrade from Node version 6 (which has reached _End of Life_ status), to Node version 8 (which is reached _Maintenance_) status.
```
Comment (DSchau) [inline docs/docs/upgrading-node-js.md]:
```suggestion
In this document you learned how you upgrade from Node version 6 (which has reached _End of Life_ status), to Node version 8 (which has reached _Maintenance_) status.
```

File: docs/docs/upgrading-node-js.md (hunk 3 above)
Comment (m-allanson) [reply docs/docs/upgrading-node-js.md]: It's a good one 👍

File: docs/docs/upgrading-node-js.md (hunk 5 above)
Comment (m-allanson) [reply docs/docs/upgrading-node-js.md]: I thought that looked wrong :)

File: docs/docs/upgrading-node-js.md (hunk 6 above)
Comment (m-allanson) [reply docs/docs/upgrading-node-js.md]: Makes sense, will update 👍 

I use `nvm` but have to refer to the docs _every_ time I use it. `n`'s API is much simpler in comparison.

File: docs/docs/upgrading-node-js.md (hunk 8 above)
Comment (m-allanson) [reply docs/docs/upgrading-node-js.md]: This has a good correction, thanks :)

**Comment by DSchau** [comment]
Oooh baby, and look at this [fresh and hot deploy preview](https://build-0aff9934-870a-4bd5-8adf-156f0b15723b.staging-previews.gtsb.io/docs/upgrading-node-js).

🔥

File: docs/docs/upgrading-node-js.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+Generally we'd recommend using [the Node version whose status is _Active LTS_](https://github.com/nodejs/Release#nodejs-release-working-group) (Node 10 at time of writing). However, in this document you'll learn how to update from Node 6 to Node 8 as this is likely to be the least disruptive upgrade for you.
+
+> What about Node 7? Stable versions of Node are evenly numbered releases - Node 6, Node 8, Node 10 etc. Only use uneven release numbers if you'd like to try cutting-edge and experimental features.
+
+There are multiple ways to update your version of Node depending on how you originally installed it. Read on to find the best approach for you.
+
+### Using Homebrew
+
+This is our recommended way to install a newer version of Node.
```
Comment (pieh) [inline docs/docs/upgrading-node-js.md]: This applies to OSX only, but it's not mentioned anywhere

Comment (m-allanson) [reply docs/docs/upgrading-node-js.md]: It's on Linux and Windows (subsystem for Linux) too! See https://www.gatsbyjs.org/tutorial/part-zero/#install-homebrew-for-nodejs

Comment (pieh) [reply docs/docs/upgrading-node-js.md]: Oh, TIL

**Comment by sidharthachatterjee** [comment]
This is now reachable via gatsby.dev/upgrading-node-js

**Pull request merged**